drivers/fastboot/fb_fsl/fb_fsl_partitions.c

```diff
@@ -239,7 +254,7 @@ static int _fastboot_parts_load_from_ptable(void)
 				/ dev_desc->blksz;
 #else
 	ptable[PTN_BOOTLOADER_INDEX].length =
-				 ANDROID_BOOTLOADER_SIZE / dev_desc->blksz;
+				 (ANDROID_BOOTLOADER_SIZE - bootloader_mmc_offset()) / dev_desc->blksz;
```

Comment:
In a way, this is a bugfix change and not related to this patch.   Maybe split out?